contracts/SignerBondsUniswapV2.sol

```diff
@@ -91,9 +95,21 @@ contract SignerBondsUniswapV2 is ISignerBondsSwapStrategy, Ownable {
         );
     }
 
-    /// @notice Swaps signer bonds.
-    /// @dev Adds incoming bonds to the overall contract balance.
-    function swapSignerBonds() external payable override {}
+    /// @notice Receive ETH upon withdrawal of risk manager's signer bonds.
+    /// @dev Do not send arbitrary funds. They will be locked forever.
+    receive() external payable {}
+
+    /// @notice Swaps the given signer bonds amount from the given risk manager.
+    /// @dev Swaps incoming signer bonds on Uniswap V2.
+    /// @param riskManager Address of the risk manager
+    /// @param amount Amount of signer bonds being swapped.
+    function swapSignerBonds(RiskManagerV1 riskManager, uint256 amount)
+        external
+        override
+    {
+        riskManager.withdrawSignerBonds(amount);
+        swapSignerBondsOnUniswapV2(riskManager, amount);
```

Comment:
Instead of passing `riskManager` to `swapSignerBondsOnUniswapV2` as it has nothing to do with the swapping logic, I'd consider performing an open auction check in this function.

Reply:
Resolved as part https://github.com/keep-network/coverage-pools/pull/88/commits/fefb57fb4ce4b9d57fbd22ee4fa6e4708b2a0640 . This function has been removed.